climakitae/core/data_interface.py

```diff
@@ -1577,3 +1586,426 @@ def _display_select(self):
         title="Choose Data Available with the Cal-Adapt Analytics Engine",
         collapsible=False,
     )
+
+
+## -------------- Data access without GUI -------------------
+
+
+def _get_user_friendly_catalog(intake_catalog, variable_descriptions):
+    """Get a user-friendly version of the intake data catalog using climakitae naming conventions"""
+
+    # Get the catalog as a dataframe
+    cat_df = intake_catalog.df.copy()
+
+    # Create new, user friendly catalog in pandas DataFrame format
+    cat_df_cleaned = pd.DataFrame()
+    cat_df_cleaned["downscaling_method"] = cat_df["activity_id"].apply(
+        lambda x: _downscaling_method_to_activity_id(x, reverse=True)
+    )
+    cat_df_cleaned["resolution"] = cat_df["grid_label"].apply(
+        lambda x: _resolution_to_gridlabel(x, reverse=True)
+    )
+    cat_df_cleaned["timescale"] = cat_df["table_id"].apply(
+        lambda x: _timescale_to_table_id(x, reverse=True)
+    )
+    cat_df_cleaned["scenario"] = cat_df["experiment_id"].apply(
+        lambda x: _scenario_to_experiment_id(x, reverse=True)
+    )
+    cat_df_cleaned["variable_id"] = cat_df["variable_id"]
+
+    # Get user-friendly variable names from variable_descriptions.csv and add to dataframe
+    cat_df_cleaned["variable"] = cat_df_cleaned.apply(
+        lambda x: _get_var_name_from_table(
+            x["variable_id"],
+            x["downscaling_method"],
+            x["timescale"],
+            variable_descriptions,
+        ),
+        axis=1,
+    )
+
+    # We dont' show the users all the available variables in the catalog
+    # These variables aren't defined in variable_descriptions.csv
+    # Here, we just remove all those variables
+    cat_df_cleaned = cat_df_cleaned[cat_df_cleaned["variable"] != "NONE"]
+
+    # Move variable column to first position
+    col = cat_df_cleaned.pop("variable")
+    cat_df_cleaned.insert(0, col.name, col)
+
+    # Remove variable_id column
+    cat_df_cleaned.pop("variable_id")
+
+    # Remove duplicate columns
```

Comment:
The code reads like rows instead of columns are being removed.

Reply:
ah... good catch!